src/lib/AutomateSDK.ts

```diff
@@ -226,6 +237,15 @@ export class AutomateSDK {
     overrides: Overrides = {},
     creatorAddress?: string
   ): Promise<CreateTaskPopulatedTransaction> {
+    //TODO: Temporary zksync fix
+    const { isDeployed } = await this.getDedicatedMsgSender();
```

Comment:
Also the `prepare` methods accepts an alternative `creatorAddress` in parameter, so you should use:
```suggestion
    const { isDeployed } = await this.getDedicatedMsgSender(creatorAddress);
```

Reply:
And you should also test `this._chainId === 324` before calling the `getDedicatedMsgSender` to avoid unnecessary rpc calls on other networks